pkg/apis/sources/v1alpha2/ping_conversion.go

```diff
@@ -19,16 +19,64 @@ package v1alpha2
 import (
 	"context"
 	"fmt"
+	"strings"
 
+	"knative.dev/eventing/pkg/apis/sources/v1beta1"
 	"knative.dev/pkg/apis"
 )
 
 // ConvertTo implements apis.Convertible
-func (source *PingSource) ConvertTo(ctx context.Context, sink apis.Convertible) error {
-	return fmt.Errorf("v1alpha2 is the highest known version, got: %T", sink)
+// Converts source from v1alpha2.PingSource into a higher version.
+func (source *PingSource) ConvertTo(ctx context.Context, obj apis.Convertible) error {
+	switch sink := obj.(type) {
+	case *v1beta1.PingSource:
+		sink.ObjectMeta = source.ObjectMeta
+		sink.Spec = v1beta1.PingSourceSpec{
+			JsonData:   source.Spec.JsonData,
+			SourceSpec: source.Spec.SourceSpec,
+		}
+		sink.Status = v1beta1.PingSourceStatus{
+			SourceStatus: source.Status.SourceStatus,
+		}
+
+		// in v1beta1, timezone has its own field
+		schedule := source.Spec.Schedule
+		if strings.HasPrefix(schedule, "TZ=") || strings.HasPrefix(schedule, "CRON_TZ=") {
+			i := strings.Index(schedule, " ")
+			eq := strings.Index(schedule, "=")
+			sink.Spec.Timezone = schedule[eq+1 : i]
+			sink.Spec.Schedule = strings.TrimSpace(schedule[i:])
+		} else {
+			sink.Spec.Schedule = schedule
+		}
+
+		return nil
+	default:
+		return fmt.Errorf("Unknown conversion, got: %T", sink)
```

Comment:
I think we have been doing something like `apis.ConvertToViaProxy(ctx, source, &v1beta1.PingSource{}, sink)`
similar thing in the ConvertFrom

Reply:
did it in v1alpha2. Thanks!